### PR TITLE
doc: update HOWTO.md

### PR DESCRIPTION
**Pull request by sarmahaj**
- update HOWTO.d file with more info on how to run manufacturing-server and manufacturing-client on their own. (when aio is not needed to be run)
- also info on SerialNumberExt option for mfg_string_type.

### CODE REVIEW

File: HOWTO.md [hunk 1]
```diff
@@ -16,10 +16,11 @@
   - `serviceinfo_api_server.yml`
 - How to run the servers
   - Manufacturing Server
+  - Manufacturing Client
```
Comment (7flying) [inline HOWTO.md:19]: The Manufacturing Client is a client, so we should move it to the `How to run the Client` section, you can add two different sections there for the `fdo-client-linuxapp` and the `fdo-manufacturing-client`

File: HOWTO.md [hunk 2]
```diff
@@ -523,10 +524,41 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
    `/etc/fdo/manufacturing-serverr.conf.d/`. The paths will be checked in that
    same order.
 
-3. Execute `fdo-manufacturing-server` or run it as a service, see sample
-   file in
+3. Execute `fdo-manufacturing-server` using below command:
+   ```
+   sudo MANUFACTURING_SERVER_CONF=/usr/share/fdo/manufacturing_server.yml LOG_LEVEL=trace ./target/debug/fdo-manufacturing-server
```
Comment (7flying) [inline HOWTO.md:529]: This command is very specific and conflicts with the pre-requisites of the HOWTO (tools built in release profile), and it also conflicts with step 2 of this section (default placement for the configuration file). 

You can add an specific section about environment variables and explain those there (`LOG_LEVEL`, `MANUFACTURING_SERVER_CONF`)

File: HOWTO.md [hunk 3]
```diff
@@ -523,10 +524,41 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+   ```
+   or run it as a service, see sample file in
    [examples/systemd](https://github.com/fedora-iot/fido-device-onboard-rs/blob/main/examples/systemd/fdo-manufacturing-server.service).
 
+### Manufacturing Client
```
Comment (7flying) [inline HOWTO.md:534]: see above, move this section to the clients section

File: HOWTO.md [hunk 4]
```diff
@@ -523,10 +524,41 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+
+1. Once you have generated required keys/certificates for the Manufacturing Server and manufacturing-server is running
+   (Refer above section: Manufacturing Server) you can run manufacturing-client. Use case such as: if you need to check
```
Comment (7flying) [inline HOWTO.md:537]: Please add links to the sections that you are referring to.

Use the standard names for the binaries when referring to them `fdo-manufacturing-server` and `fdo-manufacturing client`.

In this guide we also assume that the user has a general knowledge of the protocol, so there is no need to say that the Manufacturing server must be up and running, but it would be a nice addition. You can add a general disclaimer in the 'How to run the clients' section that the servers related to the specific client (`fdo-client-linuxapp` or `fdo-manufacturing-client`) must be up and running for the clients to work.

File: HOWTO.md [hunk 5]
```diff
@@ -523,10 +524,41 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+1. Once you have generated required keys/certificates for the Manufacturing Server and manufacturing-server is running
+   (Refer above section: Manufacturing Server) you can run manufacturing-client. Use case such as: if you need to check 
+   DI(Device Initialization)/ DIUN(Device Initialization over Untrusted Network) protocol is working as expected, then you
+   can run manufacturing-server and manufacturing-client on their own. 
+2. If you are running plainDI, set following environmental variables before running manufacturing client:
+    - DI_MFG_STRING_TYPE = SerialNumber/SerialNumberExt
+    - DI_KEY_STORAGE_TYPE = FileSystem/ Tpm 
+    - DI_SIGN_KEY_PATH = /path/to/keys
+    - DI_HMAC_KEY_PATH = /path/to/keys
+3. If you are running DIUN use below command :
```
Comment (7flying) [inline HOWTO.md:545]: add an blank line before each item in the list

File: HOWTO.md [hunk 6]
```diff
@@ -523,10 +524,41 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+   sudo MANUFACTURING_SERVER_CONF=/usr/share/fdo/manufacturing_server.yml LOG_LEVEL=trace ./target/debug/fdo-manufacturing-server
+   ```
+   or run it as a service, see sample file in
    [examples/systemd](https://github.com/fedora-iot/fido-device-onboard-rs/blob/main/examples/systemd/fdo-manufacturing-server.service).
 
+### Manufacturing Client
+
+1. Once you have generated required keys/certificates for the Manufacturing Server and manufacturing-server is running
+   (Refer above section: Manufacturing Server) you can run manufacturing-client. Use case such as: if you need to check 
+   DI(Device Initialization)/ DIUN(Device Initialization over Untrusted Network) protocol is working as expected, then you
```
Comment (7flying) [inline HOWTO.md:538]: mind spaces: DI (Device Initialization) / DIUN (Device Initialization over Untrusted Network)

File: HOWTO.md [hunk 7]
```diff
@@ -523,10 +524,41 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+   can run manufacturing-server and manufacturing-client on their own. 
+2. If you are running plainDI, set following environmental variables before running manufacturing client:
+    - DI_MFG_STRING_TYPE = SerialNumber/SerialNumberExt
+    - DI_KEY_STORAGE_TYPE = FileSystem/ Tpm 
+    - DI_SIGN_KEY_PATH = /path/to/keys
+    - DI_HMAC_KEY_PATH = /path/to/keys
+3. If you are running DIUN use below command : 
+   ```
+   sudo MANUFACTURING_SERVER_URL=http://localhost:8080 LOG_LEVEL=trace DEVICE_CREDENTIAL=dest-dir/device-credentials DIUN_PUB_KEY_ROOTCERTS=aio-dir/keys/diun_cert.pem ./target/debug/fdo-manufacturing-client
```
Comment (7flying) [inline HOWTO.md:547]: This command is very specific, we need the generic case for the HOWTO: the `device_credentials` are looked in certain paths in certain order, so we should specify those and in which order, you can explain alternatives, like providing the env var there too; the `MANUFACTURING_SERVER_URL` is also specific, we need the generic term e.g. `http://manufacturing-server` and an explanation of what it needs; `LOG_LEVEL` is not needed for it to run;  we should also explain the `USE_PLAIN_DI` env var and the relation with the `DI_MFG_STRING_TYPE` env and `DIUN_PUB_KEY_ROOTCERTS` (which also needs a generic path e.g. `/path/to/diun_cert.pem`)

Running the command like `./target/debug/fdo-manufacturing-client` conflicts with the pre-requisites of the HOWTO (see above).

Blank lines before and after code blocks.

File: HOWTO.md [hunk 8]
```diff
@@ -523,10 +524,41 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+   ```
+   In this case mfg_string_type is passed from manufacturing-server to manufacturing client during DIUN ,which comes from manufacturing-server.yml config file. You need to specify under diun settings e.g.
+   ```
+   diun:
+   mfg_string_type: SerialNumber (or) mfg_string_type: SerialNumberExt
```
Comment (7flying) [inline HOWTO.md:552]: This configuration is part of the Manufacturing server, there is already a section that explains this so you should update that section with the new `SerialNumberExt`.

File: HOWTO.md [hunk 9]
```diff
@@ -523,10 +524,41 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+   ```
+   Note: SerialNumberExt option can be used when SN(serial number) from manufacturer is not present then 'Device Info' field in device-credential file appears null. To avoid that and provide a way for device identification SerialNumberExt option is implemented.
```
Comment (7flying) [inline HOWTO.md:554]: Configuration options, field names should be formatted with ``.

File: HOWTO.md [hunk 10]
```diff
@@ -523,10 +524,41 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+   With this option ,'Device Info:' field in device-credential will be populated with a list of mac addresses of available ethernet
```
Comment (7flying) [inline HOWTO.md:555]: see above and mind spaces "With this option, "

File: HOWTO.md [hunk 11]
```diff
@@ -523,10 +524,41 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+   interfaces ,separated by semicolon.
+   e.g. 
+   ```
```
Comment (7flying) [inline HOWTO.md:558]: blank line before code block

File: HOWTO.md [hunk 12]
```diff
@@ -574,7 +606,7 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
    in
    [examples/systemd](https://github.com/fedora-iot/fido-device-onboard-rs/blob/main/examples/systemd/fdo-serviceinfo-api-server.service).
 
-## How to run the Client
+## How to run the Client (fdo-client-linuxappp)
```
Comment (7flying) [inline HOWTO.md:609]: add two sub-sectons here for the two clients

**Review by 7flying — CHANGES_REQUESTED** [review]
Thank you for the PR adding the missing info about the Manufacturing client, see the comments and also mind the commit lint error.

File: HOWTO.md (hunk 2 above)
Comment (sarmahaj) [reply HOWTO.md]: added 'Environmental Variables' section .

File: HOWTO.md [hunk 13]
```diff
@@ -19,7 +19,10 @@
   - Owner Onboarding Server
   - Rendezvous Server
   - Service Info API Server
-- How to run the Client
+- How to run the Client
```
Comment (7flying) [inline HOWTO.md:22]: Clients

File: HOWTO.md [hunk 14]
```diff
@@ -282,6 +285,24 @@ Where:
   - `plain_di`: [OPTIONAL] boolean.
   - `diun`: [OPTIONAL]
     - `mfg_string_type`: device serial number
+    Note: During `DIUN` protocol `mfg_string_type` is passed from `fdo-manufacturing-server` to
```
Comment (7flying) [inline HOWTO.md:288]: You need to align this section under `mfg_string_type`, otherwise Markdown shows it at the same level.

File: HOWTO.md [hunk 15]
```diff
@@ -282,6 +285,24 @@ Where:
   - `plain_di`: [OPTIONAL] boolean.
   - `diun`: [OPTIONAL]
     - `mfg_string_type`: device serial number
+    Note: During `DIUN` protocol `mfg_string_type` is passed from `fdo-manufacturing-server` to 
+    `fdo-manufacturing client`. 
+    This field takes values : `SerialNumber` or `StructuredDeviceInfo`.
```
Comment (7flying) [inline HOWTO.md:290]: This line is better above in:
```
- `mfg_string_type`: `SerialNumber` or `StructuredDeviceInfo`.
```
and then you follow with the explanation.

File: HOWTO.md [hunk 16]
```diff
@@ -282,6 +285,24 @@ Where:
...
+    
+    If `SerialNumber` value is provided `msg_string_type` will be populated with the device info
```
Comment (7flying) [inline HOWTO.md:292]: mfg
If the `SerialNumber` value is provided, then the `mfg_string_type`...

File: HOWTO.md [hunk 17]
```diff
@@ -282,6 +285,24 @@ Where:
...
     - `mfg_string_type`: device serial number
+    Note: During `DIUN` protocol `mfg_string_type` is passed from `fdo-manufacturing-server` to 
+    `fdo-manufacturing client`. 
+    This field takes values : `SerialNumber` or `StructuredDeviceInfo`.
+    
+    If `SerialNumber` value is provided `msg_string_type` will be populated with the device info 
+    as provide by manufacturer which is read from `/sys/devices/virtual/dmi/id/product_serial` or 
+    `/sys/devices/virtual/dmi/id/chassis_serial` locations inside the device.
+    
+    `StructuredDeviceInfo` option can be used when SN(serial number) from manufacturer is not present
```
Comment (7flying) [inline HOWTO.md:296]: when the Serial Number (SN) from the Manufacturer

File: HOWTO.md [hunk 18]
```diff
@@ -282,6 +285,24 @@ Where:
...
+    then 'Device Info' field in device-credential file appears null. To avoid that and provide a way 
+    for device identification StructuredDeviceInfo option is implemented.
```
Comment (7flying) [inline HOWTO.md:298]: `StructuredDeviceInfo`

File: HOWTO.md [hunk 19]
```diff
@@ -282,6 +285,24 @@ Where:
...
+    Note: During `DIUN` protocol `mfg_string_type` is passed from `fdo-manufacturing-server` to 
+    `fdo-manufacturing client`. 
+    This field takes values : `SerialNumber` or `StructuredDeviceInfo`.
+    
+    If `SerialNumber` value is provided `msg_string_type` will be populated with the device info 
+    as provide by manufacturer which is read from `/sys/devices/virtual/dmi/id/product_serial` or 
+    `/sys/devices/virtual/dmi/id/chassis_serial` locations inside the device.
+    
+    `StructuredDeviceInfo` option can be used when SN(serial number) from manufacturer is not present
+    then 'Device Info' field in device-credential file appears null. To avoid that and provide a way
```
Comment (7flying) [inline HOWTO.md:297]: fields with \` \` `Device Info`

File: HOWTO.md (hunk 18 above)
Comment (7flying) [reply HOWTO.md:298]: for device identification, the

File: HOWTO.md [hunk 20]
```diff
@@ -282,6 +285,24 @@ Where:
...
+    This field takes values : `SerialNumber` or `StructuredDeviceInfo`.
+    
+    If `SerialNumber` value is provided `msg_string_type` will be populated with the device info 
+    as provide by manufacturer which is read from `/sys/devices/virtual/dmi/id/product_serial` or 
+    `/sys/devices/virtual/dmi/id/chassis_serial` locations inside the device.
+    
+    `StructuredDeviceInfo` option can be used when SN(serial number) from manufacturer is not present
+    then 'Device Info' field in device-credential file appears null. To avoid that and provide a way 
+    for device identification StructuredDeviceInfo option is implemented.
+    With this option ,'Device Info:' field in device-credential will be populated with a list of mac
```
Comment (7flying) [inline HOWTO.md:299]: with this option, the  `Device Info` field in the Device Credential file

MAC

File: HOWTO.md [hunk 21]
```diff
@@ -282,6 +285,24 @@ Where:
...
+    addresses of available ethernet interfaces ,separated by semicolon.
```
Comment (7flying) [inline HOWTO.md:300]: available Ethernet interfaces, separated

File: HOWTO.md [hunk 22]
```diff
@@ -521,10 +542,14 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
    Files/manufacturing_server.yml](#manufacturing_serveryml) and place it either in
    `/usr/share/fdo`, `/etc/fdo/` or
    `/etc/fdo/manufacturing-serverr.conf.d/`. The paths will be checked in that
-   same order.
-
-3. Execute `fdo-manufacturing-server` or run it as a service, see sample
-   file in
+   same order. 
+   Mention the path for keys and certificates as generated in step1 inside `manufcturing-server.yml`.
```
Comment (7flying) [inline HOWTO.md:546]: This second line is not needed, the reader already knows how to generate keys if they have been reading the steps.

File: HOWTO.md [hunk 23]
```diff
@@ -521,10 +542,14 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+
+3. To execute `fdo-manufacturing-server` as a binary on its own use below command:
```
Comment (7flying) [inline HOWTO.md:548]: Leave it in third person `Execute fdo-manufacturing-server or run it as a service, see sample  file in [...]`

File: HOWTO.md [hunk 24]
```diff
@@ -521,10 +542,14 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+   ```
+   sudo MANUFACTURING_SERVER_CONF=/path/to/manufacturing_server.yml LOG_LEVEL=trace ./target/debug/fdo-manufacturing-server
```
Comment (7flying) [inline HOWTO.md:550]: This command is very specific and conflicts with the prerequisites of the HOWTO were the binaries have been compiled with the `-release` profile, so this command is not going to work.

Leave it as it was, the environment variables are explained below.

File: HOWTO.md [hunk 25]
```diff
@@ -576,6 +601,25 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
 
 ## How to run the Client
```
Comment (7flying) [inline HOWTO.md:602]: clients

File: HOWTO.md [hunk 26]
```diff
@@ -576,6 +601,25 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
 
+### `fdo-manufacturing-client`
+
+1. If you come across use case such as: you need to check DI (Device Initialize)
+or DIUN (Device Initialize over Untrusted Network) protocol is working as expected, 
+then you can run `fdo-manufacturing-server` and `fdo-manufacturing-client`
+on their own. 
+2. Run `fdo-manufacturing-server` as mentioned in [Manufacturing Server](#manufacturing-server).
```
Comment (7flying) [inline HOWTO.md:610]: add a blank line after every step so that it is easier to read
```
1. If you ...

2. Run ...
```

File: HOWTO.md [hunk 27]
```diff
@@ -576,6 +601,25 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
 
 ## How to run the Client
 
+### `fdo-manufacturing-client`
+
+1. If you come across use case such as: you need to check DI (Device Initialize)
```
Comment (7flying) [inline HOWTO.md:606]: I would completely delete this first step, we are in the HOWTO, not in the CONTRIBUTING guide, the info needs to be objective

File: HOWTO.md [hunk 28]
```diff
@@ -576,6 +601,25 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+2. Run `fdo-manufacturing-server` as mentioned in [Manufacturing Server](#manufacturing-server). 
+3. If you are running `plain_DI`, set following environmental variables additionally
+ before running manufacturing client:
+    - `DI_MFG_STRING_TYPE`, `DI_KEY_STORAGE_TYPE`, `DI_SIGN_KEY_PATH`, `DI_HMAC_KEY_PATH`
+Check section [Environmental Variables](#environmental-variables) for detail of each env variable.
+4. If you are running `DIUN` use below command (above step 3 is not needed): 
+   ```
+   sudo MANUFACTURING_SERVER_URL=http://localhos:8080 DEVICE_CREDENTIAL=/path/to/device-credentials 
+   DIUN_PUB_KEY_ROOTCERTS=/path/to/keys/diun_cert.pem ./path/to/binary/fdo-manufacturing-client
+   ```
```
Comment (7flying) [inline HOWTO.md:619]: Here it would be best to tell the reader that there are two ways of using this client, 1) using `USE_PLAIN_DI` (aka insecure), or 2) using a public key verification mode. Also specify how the user selects one mode or the other.

If using 1) outline the steps and the environment variables that are needed (also provide an example value of what each environment variable should take:  `DI_MFG_STRING_TYPE`, `DI_KEY_STORAGE_TYPE`, `DI_SIGN_KEY_PATH`, `DI_HMAC_KEY_PATH`);

If using 2) outline the steps and the environment variables that are needed (`DIUN_PUB_KEY_ROOTCERTS` or `DIUN_PUB_KEY_HASH`)

File: HOWTO.md [hunk 29]
```diff
@@ -576,6 +601,25 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
 ## How to run the Client
 
+### `fdo-manufacturing-client`
+
+1. If you come across use case such as: you need to check DI (Device Initialize)
+or DIUN (Device Initialize over Untrusted Network) protocol is working as expected, 
+then you can run `fdo-manufacturing-server` and `fdo-manufacturing-client`
+on their own. 
+2. Run `fdo-manufacturing-server` as mentioned in [Manufacturing Server](#manufacturing-server). 
+3. If you are running `plain_DI`, set following environmental variables additionally
```
Comment (7flying) [inline HOWTO.md:611]: environment variables

File: HOWTO.md [hunk 30]
```diff
@@ -576,6 +601,25 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+ before running manufacturing client:
+    - `DI_MFG_STRING_TYPE`, `DI_KEY_STORAGE_TYPE`, `DI_SIGN_KEY_PATH`, `DI_HMAC_KEY_PATH`
+Check section [Environmental Variables](#environmental-variables) for detail of each env variable.
+4. If you are running `DIUN` use below command (above step 3 is not needed): 
+   ```
+   sudo MANUFACTURING_SERVER_URL=http://localhos:8080 DEVICE_CREDENTIAL=/path/to/device-credentials 
+   DIUN_PUB_KEY_ROOTCERTS=/path/to/keys/diun_cert.pem ./path/to/binary/fdo-manufacturing-client
+   ```
+
```
Comment (7flying) [inline HOWTO.md:620]: You are missing a first step telling where the Device Credentials are being looked into by default.

File: HOWTO.md [hunk 31]
```diff
@@ -591,3 +635,59 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
     ```
 
 3. Run the client: `fdo-client-linuxappp`
+
+### Environmental Variables
```
Comment (7flying) [inline HOWTO.md:639]: Environment

File: HOWTO.md [hunk 32]
```diff
@@ -591,3 +635,59 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+
+Here is a list of environmental variables, their meaning and suggested values. 
+Note: Contributors/Developers are encouraged to amend to this list of env variables 
+as they come across or write new variables.
+
+1. `LOG_LEVEL`
```
Comment (7flying) [inline HOWTO.md:645]: This is Rust specific, no need to add it in the HOWTO

File: HOWTO.md [hunk 33]
```diff
@@ -591,3 +635,59 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+Note: Contributors/Developers are encouraged to amend to this list of env variables 
+as they come across or write new variables.
+
+1. `LOG_LEVEL` 
+- Usage: This variable is used to provide the log level while running any of the 
+servers or client binary or systemd service.
+- Suggested values:
+`trace` , `info`, `debug`, `error`
+
+2. `MANUFACTURING_SERVER_CONF`
```
Comment (7flying) [inline HOWTO.md:651]: Specify in which server/client this env variable is applied

File: HOWTO.md [hunk 34]
```diff
@@ -591,3 +635,59 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+- Usage: This variable is used to provide path to `manufacturing_server.yml` config file
+- Suggested values: `/path/to/manufacturing_server.yml`
+
+
+3. `DI_MFG_STRING_TYPE`
```
Comment (7flying) [inline HOWTO.md:656]: Specify in which server/client this env variable is applied

File: HOWTO.md [hunk 35]
```diff
@@ -591,3 +635,59 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+
+1. `LOG_LEVEL` 
+- Usage: This variable is used to provide the log level while running any of the 
+servers or client binary or systemd service.
+- Suggested values:
+`trace` , `info`, `debug`, `error`
+
+2. `MANUFACTURING_SERVER_CONF`
+- Usage: This variable is used to provide path to `manufacturing_server.yml` config file
+- Suggested values: `/path/to/manufacturing_server.yml`
```
Comment (7flying) [inline HOWTO.md:653]: I would reword this as "Example"

File: HOWTO.md [hunk 36]
```diff
@@ -591,3 +635,59 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+
+
+3. `DI_MFG_STRING_TYPE` 
+- Usage: This variable is used to provide what type of device info is needed from the device.
+This can be used for device identification.
+As of today this variable has two values mentioned below.
```
Comment (7flying) [inline HOWTO.md:659]: This variable has two possible values: `SerialNumber` or `StructuredDeviceInfo`, and then remove the "Suggested values" line,

File: HOWTO.md [hunk 37]
```diff
@@ -591,3 +635,59 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+- Suggested values: `SerialNumber` or `StructuredDeviceInfo`
+
+Note:
```
Comment (7flying) [inline HOWTO.md:662]: Align this note  and the following explanation to the right so that this section appears under `3. DI_MFG_STRING_TYPE`

File: HOWTO.md [hunk 38]
```diff
@@ -591,3 +635,59 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+- Usage: This variable is used to provide what type of device info is needed from the device.
+This can be used for device identification.
+As of today this variable has two values mentioned below.
+- Suggested values: `SerialNumber` or `StructuredDeviceInfo`
+
+Note: 
+`SerialNumber` - provides device identification in the form of a string provided by manufacturer
+which is read from `/sys/devices/virtual/dmi/id/product_serial` or `/sys/devices/virtual/dmi/id/chassis_serial`
+locations inside the device.
+`StructuredDeviceInfo` - whereas this option allows user to get device info in the form of a string
```
Comment (7flying) [inline HOWTO.md:666]: [ allows the user / allows a user ] to get the device info

File: HOWTO.md [hunk 39]
```diff
@@ -591,3 +635,59 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+which consists of MAC addresses of the network devices. Depending upon the availability of the
+value of `SerialNumber` from manufacturer this string may also be present in this information.
```
Comment (7flying) [inline HOWTO.md:668]: `SerialNumber` from the Manufacturer

File: HOWTO.md [hunk 40]
```diff
@@ -591,3 +635,59 @@ Please mind how the configuration file must be specifically named (e.g. `-` VS
...
+
+4. `DI_KEY_STORAGE_TYPE`
+- Usage: This variable is used to specify which type of key storage medium is to be used while
+running `fdo-manufacturing-client` 
+- Suggested values: `FileSystem` or `Tpm`
```
Comment (7flying) [inline HOWTO.md:673]: This variable has two possible values: `FileSystem` or `Tpm`, and then remove the "Suggested values" line,
